docs/AUTH.md

```diff
@@ -1,36 +1,387 @@
+- [EventHandler](../here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/models/naksha/EventHandler.java)
+- [Space](../here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/models/naksha/Space.java)
+- [XyzFeature](../here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/models/geojson/implementation/XyzFeature.java)
+- [XyzCollection](../here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/models/naksha/XyzCollection.java)
 
-- **aid**: The application identifier (**appId**) and mandatory.
-- **uid**: The user identifier (**author**) and optional.
-- **urm**: The user access-rights-matrix, it is optional. If not provided, Naksha is using the default matrix from the configuration.
-- **su**: Superuser flag, if set, the user will by-pass all security checks. This is implemented by not adding the authorization handler to the event-pipelines.
+based on User's access profile supplied using following attributes as part of [NakshaContext](../here-naksha-lib-core/src/main/java/com/here/naksha/lib/core/NakshaContext.java):
 
-## Access-Matrix Parameters
+- **appId** - Mandatory
+  - The unique application identifier, requesting current operation.
+- **author** - Optional
+  - The unique identifier of a logged-in user, who is requesting current operation.
+- **urm** - Optional (but Mandatory for meaningful authorization checks)
+  - The user-rights-matrix, defines resource level access authorized for current operation. 
+- **su** - Optional
+  - Superuser flag, if set, all authorization checks will be bypassed. This is useful where one level of authorization is already performed and we like to avoid repetitive checks on recursive/internal calls.
 
-## Access-Matrix Verbs
+## 1. URM Concept
 
-Generally the following verbs are supported:
+URM (User-Rights-Matrix) follows the Map<String, Object> format as below,
+which allows to define a matrix of an **Action** against target **Resource** by specifying one/more **AttributeMaps** of zero/more **Attributes**, that should be compared to validate request authorization:
 
-- **useXXX**: Means that a resource can be used and **id**, **title**, **description** can be viewed.
-- **manageXXX**: Means that a resource can be read and modified.
-- **createXXX**: Allow the creation of a feature. 
-- **readXXX**: Allow reading of a feature.
-- **updateXXX**: Allow to update a feature.
-- **deleteXXX**: Allow to delete a feature.
-- **purgeXXX**: Allow to purge a feature, this is final deletion.
+```json
+{
+    "urm": {
+        "xyz-hub": {
+            // zero or more Actions
+            "<action>": [
+                // one or more Attribute-Map's (atleast one should match)
+                {
+                    // zero or more access Attributes (all should match)
+                    "<accessAttribute1>": "<exactValue>",
+                    "<accessAttribute2>": "<valueWithWildcard*>",
+                    "<accessAttribute3>": [  // one or more values (all should match)
```

Comment:
Actually, one of the values should match any of the Request-Rights-Matrix values of the same attribute.
Generally, in the matrix, we always define that one attribute has exactly one value.
Therefore, the reason why this is a OR is that an attribute only has one value and the AND would simply fail always. For example, assume we have "roadType" property, how could it ever be "highways" AND "road"?
The array is just a compression, it prevents that we need too many filters that repeat everything, so the optimizer can collapse them.

Reply:
It is based on following thumb rule:
* when we need OR checks against attribute, one can add multiple AttributeMaps
* when we need AND checks against same attribute, it is added as a list of values against the same Attribute

For example:
* when you have "roadType" as "highway" OR "road", you add multiple AttributeMap.
* when you have "roadType" as "highway" AND "bidirectional" you add list of values against same Attribute.

Unless, I miss something?